internal/service/elasticache/user_group_association.go

```diff
@@ -0,0 +1,158 @@
+package elasticache
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+)
+
+func ResourceUserGroupAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceUserGroupAssociationCreate,
+		Read:   resourceUserGroupAssociationRead,
+		Delete: resourceUserGroupAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"user_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceUserGroupAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).ElastiCacheConn
+
+	input := &elasticache.ModifyUserGroupInput{
+		UserGroupId:  aws.String(d.Get("user_group_id").(string)),
+		UserIdsToAdd: aws.StringSlice([]string{d.Get("user_id").(string)}),
+	}
+
+	id := userGroupAssociationID(d.Get("user_group_id").(string), d.Get("user_id").(string))
+
+	_, err := tfresource.RetryWhenNotFound(d.Timeout(schema.TimeoutCreate), func() (interface{}, error) {
+		return conn.ModifyUserGroup(input)
```

Comment:
Hopefully it will be lessened by the user group existing first. Maybe modifying is more race resilient? I lowered the timeout which should help it not race for as long if it does.